Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -3,365 +3,2701 @@
+
+Testing universal links on a static approach includes doing the following:
+
+- Checking the Associated Domains
+- Retrieving the App Site Association File
+- Checking the Link Receiver Method
+- Checking the Data Handler Method
+- Checking if the App is Calling Other App's Universal Links
+
+###### Checking the Associated Domains
+
+Universal links require the developer to add the Associated Domains entitlement and include in it a list of the domains that the app supports (no more than about 20 to 30 domains).
+
+In Xcode, go to the "Capabilities" tab and search for "Associated Domains". You can also inspect the `.entitlements` file looking for `com.apple.developer.associated-domains`. Each of the domains must be prefixed with `applinks:`, such as `applinks:www.mywebsite.com`.
+
+Here's an example from Telegram's `.entitlements` file:
+
+```xml
+    <key>com.apple.developer.associated-domains</key>
+    <array>
+        <string>applinks:telegram.me</string>
+        <string>applinks:t.me</string>
+    </array>
+```
+
+More detailed information can be found in the [archived Apple Developer Documentation](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html#//apple_ref/doc/uid/TP40016308-CH12-SW2 "Preparing Your App to Handle Universal Links").
+
+If you don't have the original source code you still can search for this (as we did in "Entitlements Embedded in the Compiled App Binary").
+
+###### Retrieving the App Site Association File
+
+Try to retrieve the `apple-app-site-association` file from the server using the associated domains you got from the previous step. This file needs to be accessible via HTTPS, without any redirects, at `https://<domain>/apple-app-site-association` or `https://<domain>/.well-known/apple-app-site-association`.
+
+You can retrieve it yourself with your browser or use this tool: <https://branch.io/resources/aasa-validator/>. After entering the domain, it will display the file. 
+
+Example from [apple.com](https://www.apple.com/.well-known/apple-app-site-association "Apple's apple-app-site-association file"):
+
+```
+{
+    "activitycontinuation": {
+    "apps": [
+        "W74U47NE8E.com.apple.store.Jolly"
+    ]
+    },
+    "applinks": {
+    "apps": [],
+    "details": [
+        {
+        "appID": "W74U47NE8E.com.apple.store.Jolly",
+        "paths": [
+            "NOT /shop/buy-iphone/*",
+            "NOT /us/shop/buy-iphone/*",
+            "/xc/*",
+            "/shop/buy-*",
+            "/shop/product/*",
+            "/shop/bag/shared_bag/*",
+            "/shop/order/list",
+            ...
+            "/today",
+            ...
+            "/shop/watch/watch-accessories",
+            "/shop/watch/watch-accessories/*",
+            "/shop/watch/bands",
+            ...
+```
+
+Note how the "paths" key implements a white- or blacklist of relative paths. Some apps, like Telegram do not implement any (`"paths": ["*"]`), allowing all possible paths.
```

Comment:
now I explain that this "blacklisting" is not a security mechanism but rather a filter. I've called it "path exclusion" to avoid confusion. I also explain why (basically because iOS does not open universal links that were not verified at installation time, the appIDs in the server must match the requesting app appID).